### DISCUSSION TITLE
Fire up Pluto.jl notebook via `ssh` tunnel

### DISCUSSION THREAD

**Discussion by navidcy** [fonsp/Pluto.jl#1300] [Questions about using Pluto] [answered]
I'm wondering whether anybody has experience setting up a Pluto.jl notebook via an `ssh` tunnel. I'd like to open a Pluto.jl notebook on the HPC (which has GPU capabilities) but run the notebook on my local browser. Any help is welcome.

**Comment by Pramodh-G** [comment, marked as answer]
Hi, I've done this, and it works like any normal tunnel.
In the Julia REPL on the server, launch Pluto as
```
using Pluto
Pluto.run(launch_browser=false)
```
Then from the client side terminal, you could do 
`ssh -NfL 1234:127.0.0.1:1234 <username>@hostname`
Then open localhost:1234 on your browser

**Comment by gianmariomanca** [comment]
Is there an option to select a different port when launching Pluto.run? (instead of 1234)

**Reply by gianmariomanca** [reply]
OK, I guessed it and it worked: Pluto.run(port=31808)

**Comment by gianmariomanca** [comment]
@fonsp 
Is there a programmatic way to get the secret (example: http://localhost:1234/?secret=R8sEmFW8) of the pluto session running on the remote machine (to avoid having to copy-paste it manually from the remote shell to the local browser) ?

**Reply by fonsp** [reply]
No, but you can set the secret manually (and use another mechanism to generate and get the secret)

**Reply by gianmariomanca** [reply]
Great, how do I set the secret manually? I didn't notice any special keyword to manage the secret in Pluto.run(), can you attach an example with the secret set to the string "abcd". (No worries I'll pick something better for the real case :) )

**Reply by fonsp** [reply]
```julia
julia> s = Pluto.ServerSession(; options=Pluto.Configuration.from_flat_kwargs(;port=4567), secret="asdf");

julia> Pluto.run(s)

Opening http://localhost:4567/?secret=asdf in your default browser... ~ have fun!

Press Ctrl+C in this terminal to stop Pluto

```

**Reply by gianmariomanca** [reply]
Thank you! I'll try it out.

**Reply by Pramodh-G** [reply]
For me, Ctrl + click on the link opened it in my local browser, without needing to copy-paste the link
